index.js

```diff
@@ -26,7 +26,7 @@ function _fillMemDuplex(memDuplexes, index, dataRetrievalFn, response, logger){
         (err, readable) => {
         if(err){
             logger.error('failed to get full object');
-            return response.end();
+            throw new Error('failed to get full object');
```

Comment:
Isn't it better to introduce a callback and return the error instead of throwing the error? Throwing an error while being async will create another set of problems.